From: Hugo Grostabussiat <bonstra@bonstra.fr.eu.org>
Date: Sun, 28 Sep 2014 03:05:41 +0200
Subject: cryptsetup: Fix timeout on dm device.

Fix a bug in systemd-cryptsetup-generator which caused the drop-in
setting the job timeout for the dm device unit to be written with a
name different than the unit name.

https://bugs.freedesktop.org/show_bug.cgi?id=84409

(cherry-picked from commit a6fb0dc138d4e7895f8e607493279dbe4df117a1)
---
 src/cryptsetup/cryptsetup-generator.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/cryptsetup/cryptsetup-generator.c b/src/cryptsetup/cryptsetup-generator.c
index 3233e15..de91a6f 100644
--- a/src/cryptsetup/cryptsetup-generator.c
+++ b/src/cryptsetup/cryptsetup-generator.c
@@ -246,7 +246,12 @@ static int create_disk(
         }
 
         if (!noauto && !nofail) {
-                r = write_drop_in(arg_dest, name, 90, "device-timeout",
+                _cleanup_free_ char *dmname;
+                dmname = strjoin("dev-mapper-", e, ".device", NULL);
+                if (!dmname)
+                        return log_oom();
+
+                r = write_drop_in(arg_dest, dmname, 90, "device-timeout",
                                   "# Automatically generated by systemd-cryptsetup-generator \n\n"
                                   "[Unit]\nJobTimeoutSec=0");
                 if (r < 0) {
